test/upgrader_test.dart

```diff
@@ -453,17 +455,96 @@ void main() {
+  group('initialize', () {
+    test('should use fake Appcast', () async {
+      final fakeAppcast = FakeAppcast();
+      final client = MockClient.setupMockClient();
+      final upgrader = Upgrader()
+        ..client = client
+        ..appcastConfig = fakeAppcast.config
+        ..debugLogging = true
+        ..appcast = fakeAppcast
+        ..installPackageInfo(
+          packageInfo: PackageInfo(
+            appName: 'Upgrader',
+            packageName: 'com.larryaasen.upgrader',
+            version: '1.9.6',
+            buildNumber: '42',
+          ),
+        );
+
+      await upgrader.initialize();
 
-    expect(upgrader.shouldDisplayUpgrade(), false);
-    upgrader.debugDisplayAlways = true;
-    expect(upgrader.shouldDisplayUpgrade(), true);
-    upgrader.debugDisplayAlways = false;
-    expect(upgrader.shouldDisplayUpgrade(), false);
+      expect(fakeAppcast.callCount, greaterThan(0));
+    });
+  });
+
+  group('shouldDisplayUpgrade', () {
+    test('should respect `debugDisplayAlways` property', () {
+      final client = MockClient.setupMockClient();
+      final upgrader = Upgrader()
+        ..client = client
+        ..debugLogging = true;
+
+      expect(upgrader.shouldDisplayUpgrade(), false);
+      upgrader.debugDisplayAlways = true;
+      expect(upgrader.shouldDisplayUpgrade(), true);
+      upgrader.debugDisplayAlways = false;
+      expect(upgrader.shouldDisplayUpgrade(), false);
+    });
+
+    test('should be blocked when version is below minAppVersion', () async {
+      final upgrader = Upgrader()
+        ..client = MockClient.setupMockClient()
+        ..debugLogging = true
+        ..minAppVersion = '2.0.0'
+        ..installPackageInfo(
+          packageInfo: PackageInfo(
+            appName: 'Upgrader',
+            packageName: 'com.larryaasen.upgrader',
+            version: '1.9.6',
+            buildNumber: '42',
+          ),
+        );
+
+      await upgrader.initialize();
+
+      final shouldDisplayUpgrade = upgrader.shouldDisplayUpgrade();
+
+      expect(shouldDisplayUpgrade, true);
+    });
+
+    test('should be blocked when bestItem has critical update', () async {
+      final appcast = MockAppcast();
+      const version = '2.0.0';
+
+      when(appcast.bestItem()).thenReturn(
+        AppcastItem(
+          versionString: version,
+          fileURL: 'https://some.fakewebsite.com',
+          tags: [AppcastConstants.ElementCriticalUpdate],
+        ),
+      );
+
+      final upgrader = Upgrader()
+        ..client = MockClient.setupMockClient()
+        ..appcast = appcast
+        ..debugLogging = true
+        ..installPackageInfo(
+          packageInfo: PackageInfo(
+            appName: 'Upgrader',
+            packageName: 'com.larryaasen.upgrader',
+            version: version,
+            buildNumber: '42',
+          ),
+        );
+
+      await upgrader.initialize();
+
+      final shouldDisplayUpgrade = upgrader.shouldDisplayUpgrade();
```

Comment:
I think this test shows there is a logic error because the best Appcast item has the same version as the installed app. The upgrade message should not be displayed, even if it is a critical update.

Reply:
This function doesn't take the Appcast into account. I'll remove the Appcast.

Reply:
I don't believe there's a bug here